File: app/src/main/java/com/example/realtimerecycleview/MyAdapter.java
package com.example.realtimerecycleview;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyViewHolder> {

    ArrayList<User> list;
    private Context context;


    public MyAdapter(Context context, ArrayList<User> list) {

        this.context = context;

        this.list = list;

    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item,parent,false);
        return  new MyViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {

        User user = list.get(position);
        holder.Product.setText(user.getProduct());
        holder.name1.setText(user.getName());
        holder.mobile1.setText(user.getMobile());
        holder.address1.setText(user.getAddress());


        holder.amount.setText(user.getAmount());

        holder.quantity.setText(user.getQuantity());







    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public void searchDataList(ArrayList<User> searchList){
        list = searchList;
        notifyDataSetChanged();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder{

        TextView Product,amount,name1,mobile1,address1,quantity;
        CardView card;


        public MyViewHolder(@NonNull View itemView) {
            super(itemView);

            Product = itemView.findViewById(R.id.tvfirstName);

            amount = itemView.findViewById(R.id.tvage);

            name1 = itemView.findViewById(R.id.nameid);

            mobile1 = itemView.findViewById(R.id.mobileid);

            address1 = itemView.findViewById(R.id.addressid);
            quantity = itemView.findViewById(R.id.tvqty);

        }
    }

}

File: app/src/main/java/com/example/realtimerecycleview/MyAdapter.kt
package com.example.realtimerecycleview

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import com.example.realtimerecycleview.MyAdapter.MyViewHolder
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.TextView
import androidx.cardview.widget.CardView
import java.util.ArrayList

class MyAdapter(private val context: Context, var list: ArrayList<User?>?) :
    RecyclerView.Adapter<MyViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item, parent, false)
        return MyViewHolder(v)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val user = list[position]
        holder.Product.text = user.product
        holder.name1.text = user.name
        holder.mobile1.text = user.mobile
        holder.address1.text = user.address
        holder.amount.text = user.amount
        holder.quantity.text = user.quantity
    }

    override fun getItemCount(): Int {
        return list.size
    }

    fun searchDataList(searchList: ArrayList<User?>) {
        list = searchList
        notifyDataSetChanged()
    }

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var Product: TextView
        var amount: TextView
        var name1: TextView
        var mobile1: TextView
        var address1: TextView
        var quantity: TextView
        var card: CardView? = null

        init {
            Product = itemView.findViewById(R.id.tvfirstName)
            amount = itemView.findViewById(R.id.tvage)
            name1 = itemView.findViewById(R.id.nameid)
            mobile1 = itemView.findViewById(R.id.mobileid)
            address1 = itemView.findViewById(R.id.addressid)
            quantity = itemView.findViewById(R.id.tvqty)
        }
    }
}